hack/restructure_file.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+def hasbody(list1):
```

Comment:
This is regarding both `hasbody` and `haserror` functions, can you use more descriptive variable/parameter names.

And can you name the functions using pythons convention of underscore like `my_function`